### ISSUE TITLE
Offsetting surfaces

### ISSUE THREAD

**Issue by dt99jay** [mikedh/trimesh#805]
Hello – I was wondering if Trimesh currently has (or might have in the future) a method of offsetting a surface, like [Blender's Solidify modifier](https://docs.blender.org/api/current/bpy.types.SolidifyModifier.html) or a [3D version of Shapely's Buffer](https://shapely.readthedocs.io/en/latest/manual.html#object.buffer)? I'm porting a 3D printing script away from having Blender as a dependency but having trouble finding a robust replacement for Solidify... Thanks!

**Comment by science-code** [comment]
I support this question.

**Comment by mikedh** [comment]
Agreed this would be sweet. Also quite difficult to implement well hahah, PR's would be super welcome!

**Comment by drlukeparry** [comment]
Doing this with triangular meshes is not trivial. 

The most straightforward but far from perfect approach is a levelset or distance field transformation discretised into a voxel SDF field. From here the surface can be re-created from the implicit field, but it makes a very triangle heavy mesh during iso-surface extraction.

Mesh only solutions are not trivial as pointed out and are usually far from ideal.

There are some papers and code that might be of interest to explore for alternatives to the implicit approach.

https://www.jdumas.org/publication/2019/voroffset/
https://github.com/geometryprocessing/voroffset

**Comment by Marviel** [comment]
https://pypi.org/project/pyopenvdb/


This is probably what you want, though it is only available on some
platforms. It's a wrapper around Dreamworks' OpenVDB library, which uses a
fast version of the levelset method described above.

I will submit a PR when I have the time.

**Comment by clarkyeah** [comment]
@Marviel ,excellent library!

**Comment by Marviel** [comment]
No problem. I'm sorry I haven't yet submitted the promised PR, but if you
find the levelset method in openvdb, it's relatively trivial to do yourself.

Again just keep in mind -- only works on some platforms.
